packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -118,3 +126,57 @@ export const useLocalVideoStartTrigger = (isLocalVideoAvailable: boolean, should
     }
   }, [shouldTransition, isButtonStatusSynced, isPreviewCameraOn, isLocalVideoAvailable, mediaGalleryHandlers]);
 };
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ */
+const useParticipantChangedAnnouncement = (): string => {
+  const adapter = useAdapter();
+  const locale = useLocale().strings.call;
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  /**
+   * state to track whether there is currently a timer set in the MediaGallery
+   */
+  const [timeoutState, setTimeoutState] = useState<ReturnType<typeof setTimeout>>();
+
+  const setParticipantEventString = (string: string): void => {
+    setAnnouncerString('');
+    /**
+     * These set timeouts are needed to clear the announcer string in case we have multiple
+     * participants join. Since the narrator will only announce the string in the
+     * Announcer component should the string change.
+     */
+    if (timeoutState) {
+      clearTimeout(timeoutState);
+      setTimeoutState(undefined);
+    }
+    setTimeoutState(
+      setTimeout(() => {
+        setAnnouncerString(string);
+        setTimeoutState(undefined);
```

Comment:
Pretty nifty!

Reply:
@prprabhu-ms is responsible for this wizardry! 😊